packages/jsapi-utils/src/MessageUtils.test.ts

```diff
@@ -80,3 +81,78 @@ describe('requestParentResponse', () => {
     jest.useRealTimers();
   });
 });
+
+describe('requestParentResponse with parent', () => {
+  let addListenerSpy: jest.SpyInstance;
+  let removeListenerSpy: jest.SpyInstance;
+  let windowParentSpy: jest.SpyInstance;
+  let listenerCallback;
+  let messageId;
+  const mockPostMessage = jest.fn((data: Message<unknown>) => {
+    messageId = data.id;
+  });
+  beforeEach(() => {
+    addListenerSpy = jest
+      .spyOn(window, 'addEventListener')
+      .mockImplementation((event, cb) => {
+        listenerCallback = cb;
+      });
+    removeListenerSpy = jest.spyOn(window, 'removeEventListener');
+    windowParentSpy = jest.spyOn(window, 'parent', 'get').mockReturnValue(
+      TestUtils.createMockProxy<Window>({
+        postMessage: mockPostMessage,
+      })
+    );
+  });
+
+  afterEach(() => {
+    addListenerSpy.mockRestore();
+    removeListenerSpy.mockRestore();
+    windowParentSpy.mockRestore();
+    mockPostMessage.mockClear();
+    messageId = undefined;
+  });
+
+  it('Posts message to parent and subscribes to response', async () => {
```

Comment:
Are these `it` blocks the same as above? Can we maybe reuse them? wrap in a function or something.

Reply:
Good call, refactored